tx/textual/valuerenderer/dec.go

```diff
@@ -27,5 +29,41 @@ func (vr decValueRenderer) Format(_ context.Context, v protoreflect.Value) ([]Sc
 }
 
 func (vr decValueRenderer) Parse(_ context.Context, screens []Screen) (protoreflect.Value, error) {
-	panic("implement me")
+	if len(screens) != 1 {
+		return protoreflect.Value{}, fmt.Errorf("expected single screen: %v", screens)
+	}
+
+	parsed, err := parseDec(screens[0].Text)
+	if err != nil {
+		return protoreflect.Value{}, err
+	}
+
+	return protoreflect.ValueOfString(parsed), nil
+}
+
+func parseDec(v string) (string, error) {
+	parts := strings.Split(v, ".")
+	if len(parts) > 2 {
+		return "", fmt.Errorf("invalid decimal: too many points in %s", v)
+	}
+
+	intPart, err := parseInt(parts[0])
+	if err != nil {
+		return "", err
+	}
+
+	if len(parts) == 1 {
+		return intPart, nil
+	}
+
+	decPart := strings.TrimRight(parts[1], "0")
+	if len(decPart) == 0 {
+		return intPart, nil
+	}
```

Comment:
```suggestion
```

same

Reply:
If we remove this few tests are getting failing
100 != 100.0000......

Reply:
Let's do the same as for int: in tests, we check for math.LegacyDec equality